flight/Modules/OnScreenDisplay/onscreendisplay.c

```diff
@@ -727,12 +727,16 @@ void draw_alarms(int x, int y, int xs, int ys, int va, int ha, int flags, int fo
 	if (PIOS_Thread_Systime() < BOOT_DISPLAY_TIME_MS) {
 		const char *boot_reason = AlarmBootReason(alarm.RebootCause);
 		strncpy((char*)buf, boot_reason, sizeof(buf));
-		buf[strlen(boot_reason)] = '\0';
+		buf[sizeof(buf) - 2] = '\0';
 		pos = strlen(boot_reason);
 		buf[pos++] = ' ';
 	}
 
 	uint8_t state;
+
+	// With the above arrangement, can pass a length of 1 to this if
+	// there's an impossibly long boot reason.
+	// which then does the right thing and just fills it with a null.
 	int32_t len = AlarmString(&alarm, buf + pos, sizeof(buf) - 1 - pos,
```

Comment:
This will also write past the end if boot_reason is more than 99 chars (pos = strlen(boot_reason)). Also, size_t is unsigned so this could go very badly if (pos + 1) is bigger than sizeof(buf)... https://github.com/d-ronin/dRonin/blob/b323ecce4bc16b80aa638fdff07793a872d2e3f7/flight/Libraries/alarms.c#L329